### PR TITLE
RAD-320 Add date to RadiologyOrderSearchCriteria

### PR DESCRIPTION
**Pull request by bgeVam**
## Description
* Add from effective start dato to RadiologyOrderSearchCriteria, including tests
* Add to effective start dato to RadiologyOrderSearchCriteria, including tests

## Related Issue
see https://issues.openmrs.org/browse/RAD-320

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/radiology/order/HibernateRadiologyOrderDAO.java [hunk 1]
```diff
@@ -130,6 +132,23 @@ public RadiologyOrder getRadiologyOrderByUuid(String uuid) {
         if (searchCriteria.getUrgency() != null) {
             crit.add(Restrictions.eq("urgency", searchCriteria.getUrgency()));
         }
+        if (searchCriteria.getFromEffectiveStartDate() != null) {
+            Disjunction disjunction = Restrictions.disjunction();
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/module/radiology/order/HibernateRadiologyOrderDAO.java]: @bgeVam please use final on both temp vars :)

File: api/src/main/java/org/openmrs/module/radiology/order/HibernateRadiologyOrderDAO.java [hunk 2]
```diff
@@ -130,6 +132,27 @@ public RadiologyOrder getRadiologyOrderByUuid(String uuid) {
...
+            disjunction.add(Restrictions.conjunction()
+                    .add(Restrictions.eq("urgency", Urgency.ON_SCHEDULED_DATE))
+                    .add(Restrictions.ge("scheduledDate", searchCriteria.getFromEffectiveStartDate())));
+            disjunction.add(Restrictions.conjunction()
+                    .add(Restrictions.and(Restrictions.not((Restrictions.eq("urgency", Urgency.ON_SCHEDULED_DATE))))
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/module/radiology/order/HibernateRadiologyOrderDAO.java]: @bgeVam is this inner Restrictions.and() needed? or does it also work with just `Restrictions.not((Restrictions.eq("urgency", Urgency.ON_SCHEDULED_DATE)))` since its in the conjunction already

File: api/src/main/java/org/openmrs/module/radiology/order/HibernateRadiologyOrderDAO.java [hunk 3]
```diff
@@ -130,6 +132,27 @@ public RadiologyOrder getRadiologyOrderByUuid(String uuid) {
...
+            crit.add(disjunction);
+        }
+        
+        if (searchCriteria.getToEffectiveStartDate() != null) {
+            Disjunction disjunction = Restrictions.disjunction();
+            disjunction.add(Restrictions.conjunction()
+                    .add(Restrictions.eq("urgency", Urgency.ON_SCHEDULED_DATE))
+                    .add(Restrictions.le("scheduledDate", searchCriteria.getToEffectiveStartDate())));
+            disjunction.add(Restrictions.conjunction()
+                    .add(Restrictions.and(Restrictions.not((Restrictions.eq("urgency", Urgency.ON_SCHEDULED_DATE))))
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/module/radiology/order/HibernateRadiologyOrderDAO.java]: same, if above works please remove the inner Restrictions.and()

File: api/src/test/java/org/openmrs/module/radiology/order/RadiologyOrderSearchCriteriaTest.java [hunk 4]
```diff
@@ -71,7 +79,54 @@ public void build_createANewRadiologyOrderSearchCriteriaInstanceWithUrgencyIfUrg
...
+    public void build_createANewRadiologyOrderSearchCriteriaInstanceWithFromEffectiveStartDateIfFromEffectiveStartDateIsSet()
+            throws Exception {
+        
+        DateFormat format = new SimpleDateFormat("yyyy-MM-dd");
+        Date fromEffectiveStartDate = format.parse("2016-05-01");
+        radiologyOrderSearchCriteria =
+                new RadiologyOrderSearchCriteria.Builder().withFromEffectiveStartDate(fromEffectiveStartDate)
+                        .build();
+        
+        assertTrue(radiologyOrderSearchCriteria.getFromEffectiveStartDate()
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/module/radiology/order/RadiologyOrderSearchCriteriaTest.java]: @bgeVam assertThat(,is())

File: api/src/test/java/org/openmrs/module/radiology/order/RadiologyOrderSearchCriteriaTest.java [hunk 5]
```diff
@@ -71,7 +79,54 @@ public void build_createANewRadiologyOrderSearchCriteriaInstanceWithUrgencyIfUrg
...
+    public void build_createANewRadiologyOrderSearchCriteriaInstanceWithToEffectiveStartDateIfToEffectiveStartDateIsSet()
+            throws Exception {
+        
+        DateFormat format = new SimpleDateFormat("yyyy-MM-dd");
+        Date toEffectiveStartDate = format.parse("2016-05-01");
+        radiologyOrderSearchCriteria =
+                new RadiologyOrderSearchCriteria.Builder().withToEffectiveStartDate(toEffectiveStartDate)
+                        .build();
+        
+        assertTrue(radiologyOrderSearchCriteria.getToEffectiveStartDate()
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/module/radiology/order/RadiologyOrderSearchCriteriaTest.java]: @bgeVam assertThat(,is())

File: api/src/test/java/org/openmrs/module/radiology/order/RadiologyOrderServiceComponentTest.java [hunk 6]
```diff
@@ -588,14 +592,16 @@ public void getRadiologyOrders_shouldReturnAllRadiologyOrdersIncludingVoidedMatc
...
+     * @see RadiologyOrderService#getRadiologyOrders(RadiologyOrderSearchCriteria)
+     * @verifies return all radiology orders for given urgency
+     */
     @Test
     public void getRadiologyOrders_shouldReturnAllRadiologyOrdersForGivenUrgency() throws Exception {
         
+        Patient patient = patientService.getPatient(PATIENT_ID_WITH_ONE_VOIDED_AND_TWO_NON_VOIDED_RADIOLOGY_ORDERS);
         RadiologyOrderSearchCriteria radiologyOrderSearchCriteria =
-                new RadiologyOrderSearchCriteria.Builder().withUrgency(Urgency.STAT)
+                new RadiologyOrderSearchCriteria.Builder().withPatient(patient)
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/module/radiology/order/RadiologyOrderServiceComponentTest.java]: @bgeVam why did you add patient ?

Comment (bgeVam) [inline api/src/test/java/org/openmrs/module/radiology/order/RadiologyOrderServiceComponentTest.java]: new orders with this urgency got added to a different patient. test needed to be changed anyway, another possibility is to increase the number of returned orders, as a filter on urgency is a cross-patient-filter

File: api/src/main/java/org/openmrs/module/radiology/order/HibernateRadiologyOrderDAO.java [hunk 7]
```diff
@@ -130,6 +132,27 @@ public RadiologyOrder getRadiologyOrderByUuid(String uuid) {
...
         }
+        if (searchCriteria.getFromEffectiveStartDate() != null) {
+            final Disjunction disjunction = Restrictions.disjunction();
+            disjunction.add(Restrictions.conjunction()
+                    .add(Restrictions.eq("urgency", Urgency.ON_SCHEDULED_DATE))
+                    .add(Restrictions.ge("scheduledDate", searchCriteria.getFromEffectiveStartDate())));
+            disjunction.add(Restrictions.conjunction()
+                    .add(Restrictions.not(Restrictions.eq("urgency", Urgency.ON_SCHEDULED_DATE)))
+                    .add(Restrictions.ge("dateActivated", searchCriteria.getFromEffectiveStartDate())));
+            crit.add(disjunction);
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/module/radiology/order/HibernateRadiologyOrderDAO.java]: @bgeVam :cupid: :sparkling_heart: :dizzy: soo readable :+1:
